### ISSUE TITLE
Consistent errors now after upgrading

### ISSUE THREAD

**Issue by phellarv** [mgejke/hass-sectoralarm#14]
0.93.x and 0.94

2019-06-06 20:57:39 ERROR (MainThread) [homeassistant.setup] Error during setup of component sector_alarm
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 153, in _async_setup_component
    hass, processed_config)
  File "/config/custom_components/sector_alarm/__init__.py", line 66, in async_setup
    await sector_data.update()
  File "/config/custom_components/sector_alarm/__init__.py", line 125, in update
    *[task() for task in self._update_tasks])
  File "/config/custom_components/sector_alarm/__init__.py", line 154, in _update_temperatures
    for temperature in temperatures
  File "/config/custom_components/sector_alarm/__init__.py", line 154, in <dictcomp>
    for temperature in temperatures
TypeError: string indices must be integers

**Comment by mattiasbrandt** [comment]
I'm getting almost exactly the same.
Hassio version: 0.93.2 

Error during setup of component sector_alarm
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/homeassistant/setup.py", line 153, in _async_setup_component
    hass, processed_config)
  File "/config/custom_components/sector_alarm/__init__.py", line 66, in async_setup
    await sector_data.update()
  File "/config/custom_components/sector_alarm/__init__.py", line 125, in update
    *[task() for task in self._update_tasks])
  File "/config/custom_components/sector_alarm/__init__.py", line 137, in _update_status
    panel = status.get('Panel', None)
AttributeError: 'str' object has no attribute 'get'

**Comment by Pretagonist** [comment]
I checked the debug logs and get:
`2019-06-07 10:52:35 DEBUG (MainThread) [custom_components.sector_alarm] Fetched temperatures: INVALID_VERSION`

The error is like the first comment:

```
Error during setup of component sector_alarm
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 153, in _async_setup_component
    hass, processed_config)
  File "/config/custom_components/sector_alarm/__init__.py", line 66, in async_setup
    await sector_data.update()
  File "/config/custom_components/sector_alarm/__init__.py", line 125, in update
    *[task() for task in self._update_tasks])
  File "/config/custom_components/sector_alarm/__init__.py", line 154, in _update_temperatures
    for temperature in temperatures
  File "/config/custom_components/sector_alarm/__init__.py", line 154, in <dictcomp>
    for temperature in temperatures
TypeError: string indices must be integers
```

I still get data from recent arms and disarms so the component is talking to sector alarm api

**Comment by mgejke** [comment]
Haven't been able to test it myself, but hopefully it's fixed now thanks to @Pretagonist !

**Comment by Lintan83** [comment]
I get this error and the component doesn´t load.

Anyone have a suggestion?

2019-06-25 08:10:06 ERROR (MainThread) [homeassistant.setup] Error during setup of component sector_alarm
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 153, in _async_setup_component
    hass, processed_config)
  File "/config/custom_components/sector_alarm/__init__.py", line 66, in async_setup
    await sector_data.update()
  File "/config/custom_components/sector_alarm/__init__.py", line 125, in update
    *[task() for task in self._update_tasks])
  File "/config/custom_components/sector_alarm/__init__.py", line 154, in _update_temperatures
    for temperature in temperatures
  File "/config/custom_components/sector_alarm/__init__.py", line 154, in <dictcomp>
    for temperature in temperatures
TypeError: string indices must be integers

**Comment by Rawet** [comment]
> 
> 
> I get this error and the component doesn´t load.
> I hade problems with the Sector Alarm app and getting status in HA so yesterday they changed my alarm-ID and now the component doesn´t load
> 
> Anyone have a suggestion?
> 
> 2019-06-25 08:10:06 ERROR (MainThread) [homeassistant.setup] Error during setup of component sector_alarm
> Traceback (most recent call last):
> File "/usr/src/homeassistant/homeassistant/setup.py", line 153, in _async_setup_component
> hass, processed_config)
> File "/config/custom_components/sector_alarm/**init**.py", line 66, in async_setup
> await sector_data.update()
> File "/config/custom_components/sector_alarm/**init**.py", line 125, in update
> *[task() for task in self._update_tasks])
> File "/config/custom_components/sector_alarm/**init**.py", line 154, in _update_temperatures
> for temperature in temperatures
> File "/config/custom_components/sector_alarm/**init**.py", line 154, in
> for temperature in temperatures
> TypeError: string indices must be integers

Funny, I have the same issue, had alot of issues (not connected to HASS), they changed my hardware and my ID, now it get the same error as you.

**Comment by sebastian-steinmann** [comment]
Hmm, they added versioning to the api a bit back, so just logging in with a browser and snagging the latest versionnumber should fix it, unless there is a breaking change.

But my guess is that they added the version just to make it harder to reverseengineer.

Thinking they expose the api-version in one of the early responses, so could try to make it dynamic, and mabey get a better error-message.

**Comment by Lintan83** [comment]
> Hmm, they added versioning to the api a bit back, so just logging in with a browser and snagging the latest versionnumber should fix it, unless there is a breaking change.
> 
> But my guess is that they added the version just to make it harder to reverseengineer.
> 
> Thinking they expose the api-version in one of the early responses, so could try to make it dynamic, and mabey get a better error-message.

What do you mean with snagging the latest versionnumber?

**Comment by willeraab** [comment]
I tried to update to version 1_1_67 (the version used in web api calls). Unfortunately this doesn't change the result, still doesn't work with temperatures. If you disable the temperatures the panel should work though.

Problem is that I get "INVALID _VERSION" response from the call to the API that tries to fetch the temperatures. So the code tries to treat this string result as if it was the json object with temperature results. I havn't been able to figure out how this call is different from the call made by the sector website.

**Comment by mgejke** [comment]
My system is down, waiting for a technician + I'm on vacation so I can't verify or test anything for the next 10 days...

**Comment by willeraab** [comment]
> My system is down, waiting for a technician + I'm on vacation so I can't verify or test anything for the next 10 days...

Enjoy the vacation. :-)

**Comment by willeraab** [comment]
Solved: you need to update to api "v1_1_67". I've created pull requests in [asyncsector](https://github.com/mgejke/asyncsector/pull/7) / [hass-sectoralarm](https://github.com/mgejke/hass-sectoralarm/pull/15) to fix this new version.

(My previous note that it didn't work was simply because I know nothing of python. I didn't understand that asyncsector had to get an updated version number as well. Nice learning... :)  )

**Comment by Lintan83** [comment]
> Solved: you need to update to api "v1_1_67". I've created pull requests in [asyncsector](https://github.com/mgejke/asyncsector/pull/7) / [hass-sectoralarm](https://github.com/mgejke/hass-sectoralarm/pull/15) to fix this new version.
> 
> (My previous note that it didn't work was simply because I know nothing of python. I didn't understand that asyncsector had to get an updated version number as well. Nice learning... :) )

And where do i find asyncsector

**Comment by sebastian-steinmann** [comment]
https://github.com/mgejke/asyncsector/blob/master/asyncsector/asyncsector.py#L16

**Comment by Lintan83** [comment]
> https://github.com/mgejke/asyncsector/blob/master/asyncsector/asyncsector.py#L16

I dont understand a thing of this. I have Hass.IO and have never used or installed asyncsector

**Comment by sebastian-steinmann** [comment]
Seems they dont expose the version, might be able to parse the html or something.
I'll look into it

**Comment by willeraab** [comment]
> Seems they dont expose the version, might be able to parse the html or something.
> I'll look into it

It would be possible to parse it from the / that is loaded once you have sent a login - all scripts/css that are included on that page (probably looking for _<script src="/Scripts/main.js?v1_1_67"></script>_ would be a good bet). But I'd be reluctant to auto-update this, since API version changes could mean that they have changed the endpoints that asyncsector is using. Maybe not a big issue when it comes to temperatures, but with alarms it'd feel worse.

**Comment by Lintan83** [comment]
Tryed the file Willeraab uploaded today but it didn´t help

This is the error i get in my log.

2019-07-03 07:41:31 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for sector_alarm which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you do experience issues with Home Assistant.
2019-07-03 07:41:57 ERROR (SyncWorker_9) [homeassistant.util.package] Unable to install package asyncsector>=0.1.7: ERROR: Could not find a version that satisfies the requirement asyncsector>=0.1.7 (from versions: 0.0.1, 0.0.2, 0.1.0, 0.1.1, 0.1.2, 0.1.3, 0.1.4, 0.1.5)
ERROR: No matching distribution found for asyncsector>=0.1.7
2019-07-03 07:41:57 ERROR (MainThread) [homeassistant.requirements] Not initializing sector_alarm because could not install requirement asyncsector>=0.1.7
2019-07-03 07:41:57 ERROR (MainThread) [homeassistant.setup] Setup failed for sector_alarm: Could not install all requirements.

**Comment by sebastian-steinmann** [comment]
Yeah, the PR is merged in the lib, but it's not released yet. Perhaps the versionnumber should come from this plugin, and not the lib. Makes updating easier.

**Comment by sebastian-steinmann** [comment]
https://github.com/mgejke/asyncsector/pull/10

Added a PR for moving the version out if the lib and better errorhandling.

Could then be moved as an optional config prop

**Comment by phellarv** [comment]
> My system is down, waiting for a technician + I'm on vacation so I can't verify or test anything for the next 10 days...

Up and running now?

**Comment by mgejke** [comment]
At home again, but system not up and running, hopefully today...

**Comment by mgejke** [comment]
Alright, now I have a working connection between my alarm and the app/web. I've accepted a PR from @sebastian-steinmann for asyncsector so that the version string can be configured from outside of asyncsector and updated asyncsector on pypi. 

I've updated the plugin to require the new version and that seems to bring everything online for me when testing. 

Todo: Update the plugin with a new config to specify the version string(when SA changes version without impact for the plugin).
